### ISSUE TITLE
Custom GodotClass construction impossible, due to interaction between Gd::new and base fields

### ISSUE THREAD

**Issue by chitoyuu** [godot-rust/gdextension#22]
We'd like to figure out how to handle custom construction of a `GodotClass` that requires access to the `base` object. Currently, it's unclear how the `base` field should be populated on the Rust side:

```rust
#[derive(GodotClass)]
#[class(base=Reference)]
pub struct Thing {
    // We need this to we available at all times, and do not want to write a lot of `unwrap`s
    pub options: Options,

    #[base]
    base: Base<Reference>,
}

#[derive(FromVariant)]
pub struct Options {
    // Imagine that this is something you can't default construct, and don't want to clone/drop either
    pub very_important_number: Box<i32>,
}

#[derive(GodotClass)]
#[class(base=Reference)]
pub struct ThingFactory {
    #[base]
    base: Base<Reference>,
}


#[godot_api]
impl Thing {
    #[func]
    fn do_stuff(&self) {
        do_stuff_with(&self.options) // No `unwrap` necessary
    }
}

#[godot_api]
impl ThingFactory {
    #[func]
    fn create_thing(&self, options: Options) -> Gd<Thing> {
        Gd::new(Thing {
            options,
            base: todo!("what goes in here?"),
        })
    }
}
```

Bromeon mentioned on Discord that it could be an option to allow `Option<Base<T>>` for the `base` field. It's also possible to move the base pointer into the argument position instead.

**Comment by bluenote10** [comment]
Is `Gd::new` in general capable of creating the corresponding base under the hood? If so, perhaps something like `Gd::new_with_base(|base| Thing { options, base })`  may work?

**Comment by chitoyuu** [comment]
Closed by #75?
